zilencer/views.py

```diff
@@ -1032,6 +1032,21 @@ def handle_customer_migration_from_server_to_realm(
             # especially since that should be extremely rare.
             server_plan.customer = remote_realm_customer
             server_plan.save(update_fields=["customer"])
+
+            if remote_realm_customer.stripe_customer_id is None:
```

Comment:
I think it would be better to just refuse to enter this block (i.e. put this if statement above, before we change `server_plan.customer`) -- the `stripe_customer_id` and `CustomerPlan` objects need to be moved together.